cowprotocol/services
# Driver Liquidity Configuration

This PR introduces driver liquidity configuration and adds it to the auction 🎉. Note that I repurposed the `--solver-config` file into a general `--config` file as discussed on Slack.

### Test Plan

Run the driver and see it fetch liquidity!
1. Run the baseline solver with `cargo run -p baseline`
2. Run the driver with the following configuration:
   ```toml
   [[solver]]
   endpoint = "http://localhost:7872"
   name = "baseline"
   relative-slippage = "0.3"
   absolute-slippage = "100000000000000"
   address = "0x0d2584DA2F637805071f184bCFa1268eBAe8a24A"

   [liqudity]

   [[liquidity.uniswap-v2]]
   router = "0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D"
   pool-code = "0x96e8ac4277198ff8b6f785478aa9a39f403cb768dd02cbee326c3e7da348845f"
   ```
3. See an AMM!
   ```
   2023-01-18T22:25:24.477Z DEBUG solver::liquidity_collector: got 1 AMMs
   ```
   
Note that solving doesn't quite work - there are some small issues with the driver <> baseline solver boundary that needs to be solved (PR imminent).
